src/server/qtquick/private/wquickxdgdecorationmanager.cpp

```diff
@@ -163,7 +175,13 @@ WQuickXdgDecorationManagerAttached::WQuickXdgDecorationManagerAttached(WSurface
 }
 
 bool WQuickXdgDecorationManagerAttached::serverDecorationEnabled() const {
-    return m_manager->modeBySurface(m_target) == WQuickXdgDecorationManager::PreferServerSide;
+    switch (m_manager->modeBySurface(m_target)) {
+    case WQuickXdgDecorationManager::None:
```

Comment:
The behavior of `None` should be customizable by the window manager

Reply:
No, qt app send None, but it do't draw its self。

Reply:
Yon need call `decorat->setMode(WLR_XDG_TOPLEVEL_DECORATION_V1_MODE_CLIENT_SIDE);` after updateDecorationMode